### DISCUSSION TITLE
Auto skip feature

### DISCUSSION THREAD

**Discussion by ajayanuragi** [aniyomiorg/aniyomi#1111] [Ideas]
On websites like zoro.to( now aniwatch) there is a feature where intro and outro get auto skipped. 

It will be cool to have in the app. There can be an option if somebody wants to watch intros they can just disable that option.

**Comment by EdoardoCortivo** [comment]
I don't think it is even possible, simply because the app doesn't know what is playing, for example there are anime that only on the first episode don't have an opening, or others that have the ending before because there a scene after, others that after the ending just stop and other that have a preview of the next episode. And if it is a movie? It automatically skip after 23 minutes? Or there is a percentage like when the video has reached 80% it skips? There are way too many "if". On that website the admin probably watch the video first just to place the auto skip, but that is a human job.

**Reply by ajayanuragi** [reply]
There are way too many episodes for admin to watch and manually do that.


I think it's possible to implement that auto skip functionality, all intro and outro have a certain spike in music, maybe with using audios it can be done.

In the updated version of the app, there are some anime where this functionality is implemented( Maybe I am wrong as nowadays I don't use aniyomi that much so can't say for sure)

**Reply by EdoardoCortivo** [reply]
Music spikes? I don't want to offend you, but have you really thought about it?
Tell me what happens when you watch "Bocchi the Rock" and there is a concert? Or any other movie/anime/TV series that is about music?

**Reply by ajayanuragi** [reply]
My bad you are right.
